src/safeds_stubgen/api_analyzer/_api.py

```diff
@@ -165,15 +165,17 @@ def to_dict(self) -> dict[str, Any]:
 class Class:
     id: str
     name: str
-    superclasses: list[Class]
+    superclasses: list[str]
     is_public: bool
     docstring: ClassDocstring
+    is_abstract: bool = False
     constructor: Function | None = None
     constructor_fulldocstring: str = ""
```

Comment:
What is the purpose of this field? If the constructor is provided, isn't the potential doctring stored in the `Function` object anyway?

Reply:
@lars-reimann Removed this field in the constructor and added the class property
```py
@property
def is_abstract(self) -> bool:
    return "abc.ABC" in self.superclasses
```
since we need to know if the class is an abstract class for the stubs generator

Reply:
`is_abstract` was fine. The comment was about `constructor_fulldocstring`.

Reply:
This field was created for the ticket https://github.com/Safe-DS/Library-Analyzer/issues/10. For Numpydoc we'd have to search in both the class-docstring and the constructor docstring for attribute and parameter docstring information. And since the docstrings for attributes are sometimes needed before the constructor itself is parsed in the _ast_visitor.py, we have to get the constructor docstring this way.

Reply:
I'd say we talk about this on our meeting on Thursday.